fearlessIntegrationTests/RewardPayoutsServiceTests.swift

```diff
@@ -213,6 +213,236 @@ class RewardPayoutsServiceTests: XCTestCase {
+                from: storageFacade,
+                operationManager: OperationManagerFacade.sharedManager,
+                chain: chain
+            )
+
+            runtimeService.setup()
+
+            let webSocketService = createWebSocketService(
+                storageFacade: storageFacade,
+                settings: settings
+            )
+            webSocketService.setup()
+
+            guard let engine = webSocketService.connection else {
+                XCTFail("No engine")
+                return
+            }
+
+            let factory = try fetchCoderFactory(
+                runtimeService: runtimeService,
+                storageFacade: storageFacade,
+                queue: queue
+            )
+
+            guard let activeEra = try fetchActiveEra(
+                    for: chain,
+                    storageFacade: storageFacade,
+                    codingFactory: factory,
+                    operationQueue: queue
+            ) else {
+                XCTFail("No era")
+                return
+            }
+
+            do {
+                let validatorStash = [
+                    "5HoSDmKXBXeD5HBj5haVUmWsjQEcp7Tt2QmYbCd8vrkeBK4b",
+                    "5GR6SK9cj6c9uMaqZPpkMqVK99vukoRvS68ELEU2fRJ3EcNR",
+                    "5FR5YJy3uwcEkXkRaaqsgARJ4C74V1zA8C6DRAECderYFGRk",
+                    "5CFPcUJgYgWryPaV1aYjSbTpbTLu42V32Ytw1L9rfoMAsfGh",
+                    "5DUGF2j9PsCdQ9okZfRoiCgbSeq3TUEAwuHvBQ3qjX4Nz4oR",
+                    "5FbpwTP4usJ7dCFvtwzpew6NfXhtkvZH1jY4h6UfLztyD3Ng",
+                    "5E2kbY5TTqGo6PZmZccAr4nkK6zGMSn73Bocen2gGBZGkcus",
+                    "5GNy7frYA4BwWpKwxKAFWt4eBsZ9oAvXrp9SyDj6qzJAaNzB",
+                    "5C8Pev9UHEtBgd1XKhg38U4PC49Azx8v5uphtxjWiXwmpsc7"
+                ]
+                let nominatorStashAccount = "5DEwU2U97RnBHCpfwHMDfJC7pqAdfWaPFib9wiZcr2ephSfT"
+
+                let controllers = validatorStash
+                    .compactMap { accountId in
+                        try? fetchController(
+                            forStash: accountId,
+                            chain: chain,
+                            engine: engine,
+                            codingFactory: factory,
+                            queue: queue
+                        )
+                        .compactMap(\.value)
+                    }
+                    .flatMap { $0 }
+
+                let ledgers = controllers
+                    .compactMap { controller -> DyStakingLedger in
+                        try! fetchLedger(
+                            controller: controller,
+                            chain: chain,
+                            engine: engine,
+                            codingFactory: factory,
+                            queue: queue
+                        )
+                        .first!.value!
+                    }
+
+                let eras = Set<UInt32>(activeEra-84..<activeEra) // TODO use currentEra
```

Comment:
Let's make an exercise with current era too, so we don't miss this in the service